### PR TITLE
CORE-1453: allow request types to be updated

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: api/main_docs.go [hunk 1]
```diff
@@ -81,6 +81,18 @@ type requestTypeListingWrapper struct {
 //   404: errorResponse
 //   500: errorResponse
 
+// swagger:route PATCH /request-types/{name} request-types updateRequestType
+//
+// Update a Request Type
+//
+// This endpoint updates an existing request type if a request type if it exists. The name cannot be changed, but
```
Comment (psarando) [inline api/main_docs.go]: Not sure, but maybe this was the intended doc?
```suggestion
// This endpoint updates an existing request type if a request type of it exists. The name cannot be changed, but
```

File: api/request_types.go [hunk 2]
```diff
@@ -124,6 +124,76 @@ func (a *API) RegisterRequestTypeHandler(ctx echo.Context) error {
...
+			Message: err.Error(),
+		})
+	}
+	if maximumRequestsPerUser != nil && *maximumRequestsPerUser <= 0 {
+		return ctx.JSON(http.StatusBadRequest, ErrorResponse{
+			Message: "maximum-requests-per-user must be a positive integer if specified",
+		})
+	}
+
+	// Get the maximum number of concurrent requestws per user for this request type and validate it.
```
Comment (psarando) [inline api/request_types.go]:
```suggestion
	// Get the maximum number of concurrent requests per user for this request type and validate it.
```

**Review by psarando — APPROVED** [review]
LGTM 👍 
I didn't catch any issues, other than a couple of docs typos.

File: api/main_docs.go (hunk 1 above)
Comment (slr71) [reply api/main_docs.go]: Ah, that wasn't quite what I meant, but it definitely was a tpyo. Thanks for catching it!

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**